.github/workflows/test.yml

```diff
@@ -53,129 +38,106 @@ jobs:
         with:
           name: replay-driver
           path: macOS-recordreplay.so
-  test0:
-    name: End-to-end tests (0)
-    runs-on: macos-latest
-    needs: [build-server, download-browser, download-node, download-driver]
-    timeout-minutes: 20
+  preview-branch:
+    name: Wait for Vercel Preview Branch
+    runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/download-artifact@v2
-      - uses: ./.github/actions/setup-next
-      - uses: ./.github/actions/test
+      - name: Waiting for 200 from the Vercel Preview
+        uses: patrickedqvist/wait-for-vercel-preview@v1.1.1
+        id: waitFor200
```

Comment:
Yeah, so they are slightly different takes on the problem. I never actually saw `deployment_status` work, but my understanding of that approach is:

```
-> PR gets opened
-> Build gets initiated
-> All of the initial steps are performed
-> All of the playwright tests are skipped (because the deployment isn't done yet)
-> The build finishes as passing because most of the things that would cause it to fail are skipped
... Some minutes go by...
-> Vercel finished the deployment
-> All of the jobs run again
```

This flow is:

```
-> PR gets opened
-> Build gets initiated
-> All of the initial steps are performed
-> All of the playwright tests wait until the deployment returns 200
-> All tests get run
-> The overall build is passed/failed depending on tests
```

The first approach seems kind of wasteful? Like, it basically does a bunch of stuff and then says "ya know what, this thing isn't actually ready, let's cancel and do this later. Whereas the second approach runs everything once, and waits at the correct times. I would be ok with the first process if we amended it so that it *failed* the job because the vercel deployment wasn't ready yet, and that was the first check that got run, so that we didn't waste additional CI minutes, but it also seems like that would generate a lot of noisy failures. What do you think?